util/stopper.go

```diff
@@ -97,18 +97,11 @@ func (s *Stopper) AddCloser(c Closer) {
 // Returns true if the task can be launched or false to indicate the
 // system is currently draining and the task should be refused.
 func (s *Stopper) StartTask() bool {
-	// Avoid locking when we're draining (which would deadlock
```

Comment:
No; there is no longer a risk of deadlock because Stop no longer acquires the lock until after Quiesce has finished.